Documentation: describe onboarding process (#15911)
etcd-io/etcd: Documentation/contributor-guide/community-membership.md

    - Initiating, contributing and resolving discussions (emails, GitHub issues, meetings)
    - Identifying subtle or complex issues in designs and implementation PRs
- Directly contributed to the project through implementation and / or review
- Elected by supermajority of active maintainers, election steps are like below:
  - Ask two maintainers to become a sponsor
  - Send an email with candidacy to [etcd-maintainers@googlegroups.com] asking for a vote
  - Existing maintainers vote privately
  - After votes are cast one of maintainers responds with result to the original email, either positive or negative with feedback for suggested improvement
- Open a PR and add an entry to the [MAINTAINERS] file
- Request to join to etcd-maintainer teams of [etcd organization of GitHub](https://github.com/orgs/etcd-io/teams)
- Request to join to the private slack channel for etcd maintainers on [kubernetes slack](http://slack.kubernetes.io/)

> - Subscribed to [etcd-maintainers@googlegroups.com] and [etcd-maintainers-private@googlegroups.com]

Reviewer: Think this section should be indented as it follows the steps for joining.

> - Elected by supermajority of active maintainers, election steps are like below:
>   - Ask two maintainers to become a sponsor
>   - Send an email with candidacy to [etcd-maintainers@googlegroups.com] asking for a vote
>   - Existing maintainers vote privately
>   - After votes are cast one of maintainers responds with result to the original email, either positive or negative with feedback for suggested improvement
> - Open a PR and add an entry to the [MAINTAINERS] file
> - Request to join to etcd-maintainer teams of [etcd organization of GitHub](https://github.com/orgs/etcd-io/teams)
> - Request to join to the private slack channel for etcd maintainers on [kubernetes slack](http://slack.kubernetes.io/)
> - Subscribed to [etcd-maintainers@googlegroups.com] and [etcd-maintainers-private@googlegroups.com]

Reviewer: ```suggestion
- Sponsored by by two active maintainers and elected by supermajority
    - Sponsors must be from multiple member companies to demonstrate integration across community.
- To become an maintainer send an email with your candidacy to [etcd-maintainers@googlegroups.com]
    - Ensure your sponsors are @mentioned on the email
    - Include a list of contributions representative of your work on the project.
    - Existing maintainers vote will privately and respond to the email with either acceptance or with feedback for suggested improvement.
- With your membership approved you are expected to:
  - Open a PR and add an entry to the [MAINTAINERS] file
  - Subscribe to [etcd-maintainers@googlegroups.com] and [etcd-maintainers-private@googlegroups.com]
  - Request to join to etcd-maintainer teams of [etcd organization of GitHub](https://github.com/orgs/etcd-io/teams)
  - Request to join to the private slack channel for etcd maintainers on [kubernetes slack](http://slack.kubernetes.io/)
```

> - Subscribed to [etcd-maintainers@googlegroups.com] and [etcd-maintainers-private@googlegroups.com]

Reviewer: Would also like to add:
* Requesting to join [maintainers GitHub team](https://github.com/orgs/etcd-io/teams/maintainers-etcd) or one specific for the project.
* Access to shared passwords <- we should setup a LastPass,1Password or Keybase as provided by CNCF https://www.cncf.io/services-for-projects/#tools
* Access to etcd Google Cloud projects hosting etcd images

    - Identifying subtle or complex issues in designs and implementation PRs
- Directly contributed to the project through implementation and / or review
- Sponsored by two active maintainers and elected by supermajority
    - Sponsors must be from multiple member companies to demonstrate integration across community.
- To become a maintainer send an email with your candidacy to [etcd-maintainers-private@googlegroups.com]
    - Ensure your sponsors are @mentioned on the email
    - Include a list of contributions representative of your work on the project.
    - Existing maintainers vote will privately and respond to the email with either acceptance or with feedback for suggested improvement.
- With your membership approved you are expected to:
  - Open a PR and add an entry to the [MAINTAINERS] file
  - Subscribe to [etcd-maintainers@googlegroups.com] and [etcd-maintainers-private@googlegroups.com]
  - Request to join to [etcd-maintainer teams of etcd organization of GitHub](https://github.com/orgs/etcd-io/teams/maintainers-etcd)
  - Request to join to the private slack channel for etcd maintainers on [kubernetes slack](http://slack.kubernetes.io/)

>   - Request access to etcd-development GCP project (note that this GCP project will be used for shared secret management for various services)

Reviewer: ```suggestion
  - Request access to etcd-development GCP project where we publish releases
  - Request access to passwords shared between maintainers
```

> - Subscribed to [etcd-maintainers@googlegroups.com]

Author: Removing it from `Requirements` and adding to `Onboarding process` might be good?

Reviewer: Having them combined like for members is based on K8s community document. I think it makes sense because it's a requirement to become a maintainer.

Author: I see, let me update `Requirements` for describing onboarding process.

- Elected by supermajority of active maintainers.

> ### Onboarding process

Reviewer: Please mention private maintainer Slack.

Author: Got it

> - Open a PR and add an entry to the [MAINTAINERS] file.

Reviewer: As maintainer requires a supermajority I would make an exception that this should be done privatly.
I would start from asking two maintainers to become a sponsor and sending an email with candidacy to [etcd-maintainers-private@googlegroups.com] asking for a vote. 

Existing maintainers would then vote privatly (move discussion to bbc) allowing a full transparency. After votes are cast one of maintainers would respond with result to the original email. Either positive or negative with feedback for suggested improvement.

If result is positive, candidate can file a public PR

Reviewer 2: +1 @serathius

Reviewer 3: +1 Basically makes sense to me.  

> If result is positive, candidate can file a public PR

Any of the existing maintainers can also raise a public PR after there is a agreement;

Reviewer: > Any of the existing maintainers can also raise a public PR after there is a agreement;

Sure, however I would like to follow K8s and have the process driven by candidate so it's doesn't get stuck and the candidate doesn't need to wait for maintainer creating the PR.

Author: I think the steps are good, will describe in the file. But IIUC only maintainers can send emails to [[etcd-maintainers-private@googlegroups.com](mailto:etcd-maintainers-private@googlegroups.com)], so a candidate should use [[etcd-maintainers@googlegroups.com](mailto:etcd-maintainers@googlegroups.com)]?

> - Join to [etcd-maintainers@googlegroups.com] and [etcd-maintainers-private@googlegroups.com]

Reviewer: This is already mentioned as a requirement

Author: yeah, I'll remove it

>   - Request an invitation to the existing maintainer.

Reviewer: invitation to what?

> - Join to specific teams of [etcd organization of GitHub](https://github.com/orgs/etcd-io/teams)
>   - Request an invitation to the existing maintainer.

Reviewer: ```suggestion
- Request to join to etcd-maintainer teams of [etcd organization of GitHub](https://github.com/orgs/etcd-io/teams)
```
